### PR TITLE
[CARBONDATA-3127]Fix the HiveExample & TestCarbonSerde exception

### PR DESCRIPTION
**Pull request by SteNicholas**
[CARBONDATA-3127]Hive module test case has been commented off,can' t run.
This pull request fix TestCarbonSerde test case exception with maven compile.Method called deserializeAndSerializeLazySimple of TestCarbonSerde class calls the interface called serializeStartKey which is not exsit in CarbonHiveSerDe class.Just modify replace serializeStartKey with serialize and remove comments of TestCarbonSerde.
Previous hive-integration HiveExample exists obvious bug that there is no data when select HIVE_CARBON_EXAMPLE table.I have already fixed the bug above in this pull request.Location was configured wrong after creating hive table.And block lack of detail info and footer offset when do querying hive table operation.Therefore I set detail info of block with footer offset through input split.
I made mistake when squash commits in pull request #2969 ,so I create the same pull request again.
 - [x] Any interfaces changed?
        No
 - [x] Any backward compatibility impacted?
       No
 - [x] Document update required?
       No
 - [x] Testing done
       Modify TestCarbonSerde test
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
       No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2098/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10143/

File: examples/spark2/pom.xml [hunk 1]
```diff
@@ -78,7 +83,12 @@
     <dependency>
       <groupId>org.apache.httpcomponents</groupId>
       <artifactId>httpclient</artifactId>
-      <version>${httpclient.version}</version>
+      <version>4.3.4</version>
```
Comment (xubo245) [inline examples/spark2/pom.xml]: Please use the unify version

File: examples/spark2/pom.xml [hunk 2]
```diff
@@ -78,7 +83,12 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpcore</artifactId>
+      <version>4.3-alpha1</version>
```
Comment (xubo245) [inline examples/spark2/pom.xml]: Please use the unify version

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/HiveExample.scala [hunk 3]
```diff
@@ -30,44 +33,38 @@ object HiveExample {
...
+  }
 
-    val carbonSession = SparkSession
-      .builder()
-      .master("local")
-      .appName("HiveExample")
-      .config("carbonSession.sql.warehouse.dir", warehouse).enableHiveSupport()
-      .getOrCreateCarbonSession(
-        store, metaStore_Db)
+  def exampleBody(sparkSession: SparkSession, store: String): Unit = {
```
Comment (xubo245) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/HiveExample.scala]: carbonSession is better

File: examples/spark2/src/test/scala/org/apache/carbondata/examplesCI/RunExamples.scala [hunk 4]
```diff
@@ -19,12 +19,12 @@ package org.apache.carbondata.examplesCI
 
 import org.apache.spark.sql.test.util.QueryTest
 import org.scalatest.BeforeAndAfterAll
-
 import org.apache.carbondata.examples._
 import org.apache.carbondata.core.constants.CarbonCommonConstants
 import org.apache.carbondata.core.util.CarbonProperties
 import org.apache.carbondata.examples.sdk.CarbonReaderExample
 import org.apache.carbondata.examples.sql.JavaCarbonSessionExample
+import org.apache.spark.sql.test.TestQueryExecutor
```
Comment (xubo245) [inline examples/spark2/src/test/scala/org/apache/carbondata/examplesCI/RunExamples.scala]: please move o another group

File: examples/spark2/pom.xml (hunk 1 above)
Comment (SteNicholas) [reply examples/spark2/pom.xml]: @xubo245 I have already use the unify version of http client library.

File: examples/spark2/pom.xml (hunk 2 above)
Comment (SteNicholas) [reply examples/spark2/pom.xml]: @xubo245 I have already use the unify version of http core library.

File: examples/spark2/src/test/scala/org/apache/carbondata/examplesCI/RunExamples.scala (hunk 4 above)
Comment (SteNicholas) [reply examples/spark2/src/test/scala/org/apache/carbondata/examplesCI/RunExamples.scala]: @xubo245 Yeah, I move to another group.TestCarbonSerDe is also updated as same as the way of RunExamples.scala.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1892/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2101/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1893/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10148/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2102/

**Comment by xubo245** [comment]
LGTM

**Comment by jackylk** [comment]
LGTM

**Comment by jackylk** [comment]
Thanks for contributing this fix

**Pull request closed without merging**